EJERCICIO 1.
Buscar un conjunto de datos a analizar:
Se recomienda que el conjunto de datos a analizar tenga variables numéricas y categóricas. Ya que también se analizarán. 

In [17]:
import pandas as pd
import numpy as np
from scipy import stats
import seaborn as sns

iris = sns.load_dataset('iris') 

In [18]:
iris.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [19]:
iris.describe()

sepal_length  sepal_width  petal_length  petal_width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

In [20]:
iris.describe(include = "object")

species
count      150
unique       3
top     setosa
freq        50

In [21]:
iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


EJERCICIO 2.
Extraer los siguentes estadísiticos del conjunto de datos para varias de las variables disponibles:
Media
Desviación con respecto de la media
Varianza
Desviación típica
Moda
Mediana
Media ponderada
Robustez
Coeficiente de variación 

In [29]:
media = iris["sepal_width"].mean()
desv_media = np.abs(iris["sepal_width"] - media)
varianza = np.var(iris["sepal_width"])
moda, frecuencia = stats.mode(iris["sepal_width"])
mediana = np.median(iris["sepal_width"])
w = np.ones (len(iris["sepal_width"]), dtype = np.complex64)
avg = np.average(iris["sepal_width"], weights= w)
#robustez 
coeficiente = np.std(iris["sepal_width"])/np.abs(media)

print(media)
print('---')
print(desv_media)
print("---")
print(varianza)
print('---')
print(moda)
print("---")
print(mediana)
print('---')
print(avg)
print('---')
print(coeficiente)
print('---')


3.057333333333334
---
0      0.442667
1      0.057333
2      0.142667
3      0.042667
4      0.542667
         ...   
145    0.057333
146    0.557333
147    0.057333
148    0.342667
149    0.057333
Name: sepal_width, Length: 150, dtype: float64
---
0.1887128888888887
---
[3.]
---
3.0
---
(3.0573333333333337+0j)
---
0.14208819267406048
---


In [31]:
pd.options.display.max_rows = None
print(desv_media)


0      0.442667
1      0.057333
2      0.142667
3      0.042667
4      0.542667
5      0.842667
6      0.342667
7      0.342667
8      0.157333
9      0.042667
10     0.642667
11     0.342667
12     0.057333
13     0.057333
14     0.942667
15     1.342667
16     0.842667
17     0.442667
18     0.742667
19     0.742667
20     0.342667
21     0.642667
22     0.542667
23     0.242667
24     0.342667
25     0.057333
26     0.342667
27     0.442667
28     0.342667
29     0.142667
30     0.042667
31     0.342667
32     1.042667
33     1.142667
34     0.042667
35     0.142667
36     0.442667
37     0.542667
38     0.057333
39     0.342667
40     0.442667
41     0.757333
42     0.142667
43     0.442667
44     0.742667
45     0.057333
46     0.742667
47     0.142667
48     0.642667
49     0.242667
50     0.142667
51     0.142667
52     0.042667
53     0.757333
54     0.257333
55     0.257333
56     0.242667
57     0.657333
58     0.157333
59     0.357333
60     1.057333
61     0.057333
62     0

EJERCICIO 3
Interpretación de los resultados.
Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos. 

 - MEDIA: La media de nuestros datos (3.057)

 - DESVIACION DE LA MEDIA: en general nuestros datos se encuentran próximos a la media, exceptuando de contados casos concretos en los que se aleja un poco más de la media, en estos casos destacados sería interesante estudiar el motivo de la desviación.
 
 - VARIANZA: Con la varianza (0.188) podemos comprobar que nuestros datos no son dispersos
 
 - MODA:


